src/log.cpp

```diff
@@ -53,8 +75,14 @@ void Log::debug(const char *fmt, ... )
     vsnprintf( buf, sizeof(buf) - 1, fmt, vl );
     va_end( vl );
 
-    // Add the log line
-    g()->add( buf );
+    Log * self = g();
+
+    if (self->mOutputFile == nullptr) {
```

Comment:
If we are writing into log, this has to be done in add() to avoid code duplication AND to ensure the timestamps are written to the log.

Reply:
I moved it to add() as an optional branch after writing to the buffer.